internal/domain/application_oidc_test.go

```diff
@@ -185,3 +187,489 @@ func TestApplicationValid(t *testing.T) {
+			}
+		})
+	}
+}
+
+func Test_CheckRedirectUrisImplicitAndCode(t *testing.T) {
+	type args struct {
+		appType      OIDCApplicationType
+		redirectUris []string
+	}
+	tests := []struct {
+		name string
+		want *Compliance
+		args args
+	}{
+		{
+			name: "implicit and code https",
+			want: &Compliance{
+				NoneCompliant: false,
+				Problems:      nil,
+			},
+			args: args{
+				redirectUris: []string{"https://redirect.to/me"},
+			},
+		},
+		// {
+		// 	name: "custom protocol, not native",
+		// 	want: &Compliance{
+		// 		NoneCompliant: true,
+		// 		Problems:      []string{"Application.OIDC.V1.Implicit.RedirectUris.CustomNotAllowed"},
+		// 	},
+		// 	args: args{
+		// 		redirectUris: []string{"protocol://redirect.to/me"},
+		// 		appType:      OIDCApplicationTypeWeb,
+		// 	},
+		// },
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			compliance := new(Compliance)
+
+			CheckRedirectUrisImplicitAndCode(compliance, tt.args.appType, tt.args.redirectUris)
+
+			if tt.want.NoneCompliant != compliance.NoneCompliant {
+				t.Errorf("NoneCompliant: expected: %v, got %v", tt.want.NoneCompliant, compliance.NoneCompliant)
+			}
+			if !reflect.DeepEqual(tt.want.Problems, compliance.Problems) {
+				t.Errorf("Problems: expected: %v, got %v", tt.want.Problems, compliance.Problems)
+			}
+		})
+	}
+}
+
+func TestCheckRedirectUrisImplicitAndCode(t *testing.T) {
+	type args struct {
+		appType      OIDCApplicationType
+		redirectUris []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *Compliance
+	}{
+		{
+			name: "only https",
+			args: args{},
+			want: &Compliance{},
+		},
+		{
+			name: "custom protocol not native app",
+			args: args{
+				appType:      OIDCApplicationTypeWeb,
+				redirectUris: []string{"custom://nirvana.com"},
+			},
+			want: &Compliance{
+				NoneCompliant: true,
+				Problems:      []string{"Application.OIDC.V1.Implicit.RedirectUris.CustomNotAllowed"},
+			},
+		},
+		{
+			name: "http localhost user agent app",
+			args: args{
+				appType:      OIDCApplicationTypeUserAgent,
+				redirectUris: []string{"http://localhost:9009"},
+			},
+			want: &Compliance{
+				NoneCompliant: true,
+				Problems:      []string{"Application.OIDC.V1.Code.RedirectUris.HttpOnlyForWeb"},
+			},
+		},
+		{
+			name: "http, not only localhost native app",
+			args: args{
+				appType:      OIDCApplicationTypeNative,
+				redirectUris: []string{"http://nirvana.com", "http://localhost:9009"},
+			},
+			want: &Compliance{
+				NoneCompliant: true,
+				Problems:      []string{"Application.OIDC.V1.Native.RedirectUris.MustBeHttpLocalhost"},
+			},
+		},
+		{
+			name: "not allowed combination",
+			args: args{
+				appType:      OIDCApplicationTypeNative,
+				redirectUris: []string{"https://nirvana.com", "cutom://nirvana.com"},
+			},
+			want: &Compliance{
+				NoneCompliant: false, //TODO: imo this should be true
+				Problems:      []string{"Application.OIDC.V1.NotAllCombinationsAreAllowed"},
```

Comment:
IMO https is always allowed and custom for native is ok as well

the way we check at the moment we know that it's not "not compliant" but we're not 100% sure if it is compliant in all cases... we might consider improving the checks (in the future?)